src/main/webapp/index-script.js

```diff
@@ -74,42 +73,78 @@ function searchByCoordinates(coordinate) {
   });
 }
 
-// Search using text query
-function searchByText(textQuery){
-  // Create the places service.
-
+/* Search Places API for relevant locations using text query. */
+function searchByText(textQuery) {
   // Perform a query (hard-coded to be the Googleplex for right now)
   var request = {
       query: textQuery,
       fields: ['place_id', 'geometry']
   };
 
   var service = new google.maps.places.PlacesService(map);
-
   service.findPlaceFromQuery(request, function(results, status) {
     if (status === google.maps.places.PlacesServiceStatus.OK) {
       handleSearchResults(results, service);
     }
   });
 }
 
-// Takes the results from a given search and prints them to the console
+/* Accepts results from places query and returns array of details for nearby locations. */
 function handleSearchResults(results, service) {
+  // Center on the queried location
+  if (results.length > 0) { 
+    map.setCenter(results[0].geometry.location);
+  }
+  
+  var promises = [];
   results.forEach((result)=> {
     var request = {
       placeId: result.place_id,
-      fields: ['name']
+      fields: [
+        'name',
+        'vicinity',
+        'reviews',
+        'place_id',
+        'opening_hours',
+        'geometry',
+        'icon',
+        'international_phone_number',
+        'website'
+      ]
     };
-    // Finds the exact details about the place
-    service.getDetails(request, (place, status) => {
-      if (status == google.maps.places.PlacesServiceStatus.OK) {
-        console.log(place.name);
-      }
-    })
+
+    // Creates a promise to return details from places api request.
+    const promise = new Promise((resolve, reject) => {
+      service.getDetails(request, (place, status) => {
+        if (status == google.maps.places.PlacesServiceStatus.OK) {
+          resolve(place);
+        }
+      });
+    });
+    promises.push(promise);
+    
   });
 
-  // Center on the queried location
-  if (results.length > 0) { 
-    map.setCenter(results[0].geometry.location);
-  }
+  // Waits on all promises to complete before passing the results into the next function.
+  Promise.all(promises).then(places => {
+    populateSearch(places); // Placeholder
+  });
+}
+
+// Place-holder for function that fills out search results page.
+function populateSearch(places) {
+  places = sortPlacesByRating(places);
+  console.log(places);
+  return null; 
+  // TO-DO: Implement or replace this function.
```

Comment:
Nit. Use TODO (convention) instead of TO-DO. It's also typically better to create an issue in github and link the TODO comment to it, that way you can track TODO/Issues in one fixed space without worrying about forgetting.